xml/ftp_banners.xml

```diff
@@ -881,6 +1027,246 @@ more text</example>
+    <param pos="0" name="os.product" value="Windows"/>
+  </fingerprint>
+  <fingerprint pattern="^([\w.-]+) X2 WS_FTP Server ([\d.]{3,6}\s?\(\d+\))$">
+    <description>WS_FTP FTP Server on Windows - X2 variant</description>
+    <example service.version="7.7(50012467)" host.name="a.host.name.tld">a.host.name.tld X2 WS_FTP Server 7.7(50012467)</example>
+    <example service.version="5.0.5 (1989540204)" host.name="a.host.name.tld">a.host.name.tld X2 WS_FTP Server 5.0.5 (1989540204)</example>
+    <param pos="0" name="service.family" value="WS_FTP"/>
+    <param pos="0" name="service.product" value="WS_FTP"/>
+    <param pos="0" name="service.vendor" value="Ipswitch"/>
+    <param pos="2" name="service.version"/>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.device" value="General"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="Windows"/>
+    <param pos="1" name="host.name"/>
+  </fingerprint>
+  <fingerprint pattern="^V2 WS_FTP Server ([\d.]{3,6}\s?\(\d+\))$">
+    <description>WS_FTP FTP Server on Windows - V2 variant</description>
+    <example service.version="6.1(05544322)">V2 WS_FTP Server 6.1(05544322)</example>
+    <param pos="0" name="service.family" value="WS_FTP"/>
+    <param pos="0" name="service.product" value="WS_FTP"/>
+    <param pos="0" name="service.vendor" value="Ipswitch"/>
+    <param pos="1" name="service.version"/>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.device" value="General"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="Windows"/>
+  </fingerprint>
+  <fingerprint pattern="^FTP Server \(ZyWALL (USG\s?[\w-]+)\) \[([\h:.]+)\]$">
+    <description>ZyXEL Unified Security Gateway</description>
+    <example hw.product="USG 20" host.ip="::ffff:192.168.0.2">FTP Server (ZyWALL USG 20) [::ffff:192.168.0.2]</example>
+    <example hw.product="USG100-PLUS" host.ip="::ffff:192.168.5.101">FTP Server (ZyWALL USG100-PLUS) [::ffff:192.168.5.101]</example>
+    <example hw.product="USG 20" host.ip="10.0.0.2">FTP Server (ZyWALL USG 20) [10.0.0.2]</example>
+    <param pos="0" name="service.vendor" value="ZyXEL"/>
+    <param pos="0" name="service.family" value="Unified Security Gateway"/>
+    <param pos="0" name="service.product" value="FTPD"/>
+    <param pos="2" name="host.ip"/>
+    <param pos="0" name="hw.vendor" value="ZyXEL"/>
+    <param pos="0" name="hw.family" value="Unified Security Gateway"/>
+    <param pos="1" name="hw.product"/>
+  </fingerprint>
+  <fingerprint pattern="^Welcome to TP-LINK FTP server$">
+    <description>FTPD on a TP-LINK device (no version/host info)</description>
+    <example>Welcome to TP-LINK FTP server</example>
+    <param pos="0" name="hw.vendor" value="TP-LINK"/>
+  </fingerprint>
+   <fingerprint pattern="^ucftpd\((\w{3}  \d{,2} \d{4}-\d\d:\d\d:\d\d)\) FTP server ready\.$">
```

Comment:
`\d{,2}` will match 0-2 digits.  Also, are you sure there are explicitly two spaces between the month and the day?  Maybe just use `\s+`